Formula/verilator.rb

```diff
@@ -26,22 +26,37 @@ class Verilator < Formula
   uses_from_macos "perl"
   uses_from_macos "python", since: :catalina
 
+  on_macos do
+    depends_on "gcc"
+  end
+
   skip_clean "bin" # Allows perl scripts to keep their executable flag
 
   # error: specialization of 'template<class _Tp> struct std::hash' in different namespace
   fails_with gcc: "5"
 
+  # Build hangs when using Clang
+  # https://github.com/Homebrew/homebrew-core/pull/124827
+  # https://github.com/Homebrew/homebrew-core/pull/130893
+  # https://github.com/Homebrew/homebrew-core/pull/133912
+  fails_with :clang
+
   def install
+    # V3Lexer_pregen.yy.cpp:369:10: fatal error: FlexLexer.h: No such file or directory
+    ENV.append_to_cflags "-isystem /Library/Developer/CommandLineTools/usr/include" if OS.mac?
+
     system "autoconf"
     system "./configure", "--prefix=#{prefix}"
     # `make` and `make install` need to be separate for parallel builds
     system "make"
     system "make", "install"
 
     # Avoid hardcoding build-time references that may not be valid at runtime.
+    gcc = Formula["gcc"]
+    cxx = OS.mac? ? gcc.opt_bin/"g++-#{gcc.any_installed_version.major}" : "c++"
```

Comment:
We should probably just install `cc` and `c++` symlinks for `gcc` in `libexec` or something so these don't need to be rebuilt on GCC version bumps.